### DISCUSSION TITLE
Radiomaster Boxer support in Companion

### DISCUSSION THREAD

**Discussion by ChrisBalmforth** [EdgeTX/edgetx#3100] [Q&A]
I've just taken delivery of the new Radiomaster Boxer ELRS version. This is fitted with v2.8.0 firmware but there doesn't appear to be a Boxer profile yet in Companion 2.8.0. Is there a pre-release version of Companion available which will support the Boxer?

**Comment by pfeerick** [comment]
Check the Check tab of https://github.com/EdgeTX/edgetx/pull/2910 for a build for your OS, and let me know how you go. :)

**Reply by ChrisBalmforth** [reply]
Thanks for the speedy reply! Unfortunately the linux version artifact is marked expired so I can't download it. I'm a bit out of my depth here - if I click on build will the system update the artifact or do I have to compile it myself? That's going to be beyond me I'm afraid!

**Reply by pfeerick** [reply]
Try again in about an hour... didn't realise I'd only retriggered the windows build the other day. 🤦

**Reply by ChrisBalmforth** [reply]
Many thanks Peter, that works just fine. Exactly what I wanted!

**Reply by ChrisBalmforth** [reply]
Do you want feedback on issues I discover with this release or is it likely to be out of date now? If you do want feedback which is the correct place to give it?

**Reply by pfeerick** [reply]
Probably in the PR, simply to keep the information together. PR branch is the latest version, so it's just a matter of looking at the commit log for a PR if you think you have found an issue that might have already been fixed.

**Comment by coaster007** [comment]
i am waiting for the Boxer and now i try the Boxer Companion for now.
Discovered one issue: the pots s1,s2,s3 defined in Companion don't are propagated to the Simulator.
![BoxerSimulator](https://user-images.githubusercontent.com/34204590/215838599-4f144f24-bc95-486e-899c-884d151808a2.png)
![BoxerCompanion](https://user-images.githubusercontent.com/34204590/215838604-6ceda1d2-2349-4a7e-b210-aabdafe1c5bc.png)

**Reply by ChrisBalmforth** [reply]
As pfeerick says in his summary of changes in #2910  the Simulator is not working yet. I also find that S1, S2 and S3 are labelled CYC1, CYC2 and CYC3 in this version of Companion and aren't uploaded to the Boxer when you save the model. I am still finding it very useful though, you just have to check the model after uploading it to the Boxer and make any necessary modifications. It's still far easier than trying to create models from scratch on the transmitter!

**Reply by coaster007** [reply]
thank's chris for the info, but shall we open a issue or waiting for a newer Boxer Companion Version ?

**Reply by ChrisBalmforth** [reply]
I'm not sure what the correct procedure is for a special pre-production build like this. I asked pfeerick if he wanted feedback on this build (see above) but there's been no response yet.

**Reply by pfeerick** [reply]
I had this suspicion there was still something not quite right around the pots, so thanks for confirming that. The bullet point about the simulator is no longer correct as the initial simulator support was added after receiving graphics from RM for it. Real life has just been getting in the way... 

So as mentioned above, you can leave comments in the PR (which is more appropriate than opening a new issue, since any feedback is  in relation to the work in progress rather than a released/nightly feature or bug in the release/nightly version).

**Reply by pfeerick** [reply]
Haven't looked into the code yet, but I can reproduce this - it shows CYC1-CYC3 because S1-S3 are not configured in the hardware tab (so effectively don't exist), and since the simulator build is heli enabled, they are simply the next sources in the list. If you configure S1-S2 as pot with detent, and s3 as multipos, everything should work fine.

**Reply by ChrisBalmforth** [reply]
Thanks Peter! I'll try that.

**Reply by pfeerick** [reply]
Ok, was a typo on my part - you should have a usable build in about an hour.

**Reply by ChrisBalmforth** [reply]
Brilliant! I'll try it tomorrow - it's 1am here!

**Reply by ChrisBalmforth** [reply]
I tried the Linux version of the new build and it seems to work fine on my Boxer. Pots S1 and S3 transfer OK between Companion and Tx in both directions. Thanks!

**Reply by pfeerick** [reply]
Thanks for that. Since you're actively using it, please let me know if anything else crops up in the next day or two, otherwise I'll most likely merge this as is, as the niggling issue I couldn't put my finger on (don't you just hate that... you know at the back of your mind *something* doesn't seem right, but can't place _what_ it is 😆 ) has been caught ;)

**Reply by ChrisBalmforth** [reply]
Will do!
